gr-fec/lib/fec_mtrx_impl.cc

```diff
@@ -148,39 +142,43 @@ void write_matrix_to_file(const std::string filename, matrix_sptr M)
 
 matrix_sptr generate_G_transpose(matrix_sptr H_obj)
 {
-    unsigned int k = H_obj->size1;
-    unsigned int n = H_obj->size2;
-    unsigned int row_index, col_index;
+    const unsigned int k = H_obj->size1;
+    const unsigned int n = H_obj->size2;
 
-    gsl_matrix* G_transp = gsl_matrix_alloc(n, k);
+    gsl_matrix_uchar* G_transp = gsl_matrix_uchar_alloc(n, k);
 
     // Grab P' matrix (P' denotes P transposed)
-    gsl_matrix* P_transpose = gsl_matrix_alloc(n - k, k);
-    for (row_index = 0; row_index < n - k; row_index++) {
-        for (col_index = 0; col_index < k; col_index++) {
-            int value = gsl_matrix_get((gsl_matrix*)(H_obj.get()), row_index, col_index);
-            gsl_matrix_set(P_transpose, row_index, col_index, value);
+    gsl_matrix_uchar* P_transpose = gsl_matrix_uchar_alloc(n - k, k);
+    for (unsigned int row_index = 0; row_index < n - k; row_index++) {
+        for (unsigned int col_index = 0; col_index < k; col_index++) {
+            gsl_matrix_uchar_set(P_transpose,
+                                 row_index,
+                                 col_index,
+                                 gsl_matrix_uchar_get(H_obj.get(), row_index, col_index));
         }
     }
 
     // Set G transpose matrix (used for encoding)
-    gsl_matrix_set_zero(G_transp);
-    for (row_index = 0; row_index < k; row_index++) {
-        col_index = row_index;
-        gsl_matrix_set(G_transp, row_index, col_index, 1);
+    gsl_matrix_uchar_set_zero(G_transp);
+    for (unsigned int row_index = 0; row_index < k; row_index++) {
+        const auto col_index = row_index;
+        gsl_matrix_uchar_set(G_transp, row_index, col_index, 1);
     }
```

Comment:
Nit. I would write this as follows. It makes it even more obvious that we are only accessing the diagonal.
```suggestion
    gsl_matrix_uchar_set_zero(G_transp);
    for (unsigned int row_index = 0; row_index < k; row_index++) {
        gsl_matrix_uchar_set(G_transp, row_index, row_index, 1);
    }
```
Also, it looks like all this thing could be replaced by [`gsl_matrix_set_identity`](https://www.gnu.org/software/gsl/doc/html/vectors.html#c.gsl_matrix_set_identity).

Reply:
agreed, let's use the dedicated function